test/mainnet-fork/wstETH.t.sol

```diff
@@ -0,0 +1,283 @@
+pragma solidity >=0.8.0;
+
+import "./ForkTestBase.t.sol";
+
+interface IWithdrawalQueue {
+    function getWithdrawalStatus(uint256[] calldata _requestIds)
+        external view returns (WithdrawalRequestStatus[] memory statuses);
+    function finalize(uint256 _lastRequestIdToBeFinalized, uint256 _maxShareRate) external payable;
+}
+
+struct WithdrawalRequestStatus {
+    uint256 amountOfStETH;
+    uint256 amountOfShares;
+    address owner;
+    uint256 timestamp;
+    bool    isFinalized;
+    bool    isClaimed;
+}
+
+contract MainnetControllerWstETHTestBase is ForkTestBase {
+
+    function _getBlock() internal override pure returns (uint256) {
+        return 23469772; //  September 29, 2025
+    }
+
+}
+
+contract MainnetControllerDepositToWstETHFailureTests is MainnetControllerWstETHTestBase {
+
+    function test_depositToWstETH_notRelayer() external {
+        vm.expectRevert(abi.encodeWithSignature(
+            "AccessControlUnauthorizedAccount(address,bytes32)",
+            address(this),
+            RELAYER
+        ));
+        mainnetController.depositToWstETH(1e18);
+    }
+
+    function test_depositToWstETH_zeroMaxAmount() external {
+        vm.prank(relayer);
+        vm.expectRevert("RateLimits/zero-maxAmount");
+        mainnetController.depositToWstETH(1e18);
+    }
+
+    function test_depositToWstETH_rateLimitsBoundary() external {
+        bytes32 key = RateLimitHelpers.makeAssetKey(
+            mainnetController.LIMIT_WSTETH_DEPOSIT(),
+            Ethereum.WETH
+        );
+
+        vm.prank(Ethereum.SPARK_PROXY);
+        rateLimits.setRateLimitData(key, 1_000e18, uint256(1_000e18) / 1 days);
+
+        deal(Ethereum.WETH, address(almProxy), 1_000e18);
+
+        vm.startPrank(relayer);
+        vm.expectRevert("RateLimits/rate-limit-exceeded");
+        mainnetController.depositToWstETH(1_000e18 + 1);
+
+        mainnetController.depositToWstETH(1_000e18);
+    }
+
+}
+
+contract MainnetControllerDepositToWstETHTests is MainnetControllerWstETHTestBase {
+
+    function test_depositToWstETH() external {
+        bytes32 key = RateLimitHelpers.makeAssetKey(
+            mainnetController.LIMIT_WSTETH_DEPOSIT(),
+            Ethereum.WETH
+        );
+
+        vm.prank(Ethereum.SPARK_PROXY);
+        rateLimits.setRateLimitData(key, 1_000e18, uint256(1_000e18) / 1 days);
+
+        deal(Ethereum.WETH, address(almProxy), 1_000e18);
+
+        assertEq(rateLimits.getCurrentRateLimit(key), 1_000e18);
+
+        assertEq(IERC20(Ethereum.WETH).balanceOf(address(almProxy)),   1_000e18);
+        assertEq(IERC20(Ethereum.WSTETH).balanceOf(address(almProxy)), 0);
+
+        vm.prank(relayer);
+        mainnetController.depositToWstETH(1_000e18);
+
+        assertEq(rateLimits.getCurrentRateLimit(key), 0);
+
+        assertEq(IERC20(Ethereum.WETH).balanceOf(address(almProxy)),   0);
+        assertEq(IERC20(Ethereum.WSTETH).balanceOf(address(almProxy)), 823.029395390731625220e18);
```

Comment:
Add assertion for `getStETHByWstETH(wsteth.balanceOf(almProxy))` to check that its roughly 1000

Reply:
added